beacon-chain/state/interfaces_epbs.go

```diff
@@ -0,0 +1,28 @@
+package state
+
+import (
+	"github.com/prysmaticlabs/prysm/v5/consensus-types/primitives"
+	enginev1 "github.com/prysmaticlabs/prysm/v5/proto/engine/v1"
+)
+
+type ReadOnlyEpbs interface {
+	PreviousInclusionListSlot() (primitives.Slot, error)
+	PreviousInclusionListProposer() (primitives.ValidatorIndex, error)
+	LatestInclusionListSlot() (primitives.Slot, error)
+	LatestInclusionListProposer() (primitives.ValidatorIndex, error)
+	IsParentBlockFull() (bool, error)
+	ExecutionPayloadHeader() (*enginev1.ExecutionPayloadHeaderEPBS, error)
+	LatestBlockHash() ([]byte, error)
+	LatestFullSlot() (primitives.Slot, error)
+	LastWithdrawalsRoot() ([]byte, error)
+}
+
+type WriteOnlyEpbs interface {
+	SetExecutionPayloadHeader(val *enginev1.ExecutionPayloadHeaderEPBS) error
+	UpdatePreviousInclusionListData() error
+	SetLatestInclusionListSlot(val primitives.Slot) error
+	SetLatestInclusionListProposer(val primitives.ValidatorIndex) error
+	SetLatestBlockHash(val []byte) error
+	SetLatestFullSlot(val primitives.Slot) error
+	SetLastWithdrawalsRoot(val []byte) error
+}
```

Comment:
I'd say preemptively I'd be inclined to remove the version control from these methods and avoid the error return.